src/Workspaces/CSharp/Portable/CodeGeneration/CSharpCodeGenerationService.cs

```diff
@@ -273,43 +272,35 @@ protected override TDeclarationNode AddNamespace<TDeclarationNode>(TDeclarationN
         }
 
         public override TDeclarationNode AddParameters<TDeclarationNode>(
-            TDeclarationNode destinationMember,
+            TDeclarationNode destination,
             IEnumerable<IParameterSymbol> parameters,
             CodeGenerationOptions options,
             CancellationToken cancellationToken)
         {
-            var memberDeclaration = destinationMember as MemberDeclarationSyntax;
-            if (memberDeclaration == null)
-            {
-                return destinationMember;
-            }
+            var syntaxGenerator = CSharpSyntaxGenerator.Instance;
+            var currentParameterList = CSharpSyntaxGenerator.GetParameterList(destination);
 
-            var currentParameterList = memberDeclaration.GetParameterList();
             if (currentParameterList == null)
             {
-                return destinationMember;
+                return destination;
             }
 
             var currentParamsCount = currentParameterList.Parameters.Count;
             var seenOptional = currentParamsCount > 0 && currentParameterList.Parameters[currentParamsCount - 1].Default != null;
             var isFirstParam = currentParamsCount == 0;
+            var newParams = ArrayBuilder<SyntaxNode>.GetInstance();
 
-            var parameterNodesAndTokens = currentParameterList.Parameters.GetWithSeparators().ToList();
             foreach (var parameter in parameters)
             {
-                if (parameterNodesAndTokens.Count > 0 && parameterNodesAndTokens.Last().Kind() != SyntaxKind.CommaToken)
-                {
-                    parameterNodesAndTokens.Add(SyntaxFactory.Token(SyntaxKind.CommaToken));
-                }
```

Comment:
why did you get rid of this?

Reply:
Since we're using a SyntaxGenerator now, the generator can add in the parameters to to its appropriate destination 
var finalMember = CSharpSyntaxGenerator.Instance.AddParameters(destination, newParams.ToImmutableAndFree());

Reply:
got it.  took a moment to figure this out.   this part of the change LGTM.